README.md

```diff
@@ -1266,104 +1751,20 @@
   - It's okay to create get() and set() functions, but be consistent.
 
     ```javascript
-    function Jedi(options) {
-      options || (options = {});
-      var lightsaber = options.lightsaber || 'blue';
-      this.set('lightsaber', lightsaber);
-    }
-
-    Jedi.prototype.set = function(key, val) {
-      this[key] = val;
-    };
-
-    Jedi.prototype.get = function(key) {
-      return this[key];
-    };
-    ```
-
-**[⬆ back to top](#table-of-contents)**
-
-
-## Constructors
-
-  - Assign methods to the prototype object, instead of overwriting the prototype with a new object. Overwriting the prototype makes inheritance impossible: by resetting the prototype you'll overwrite the base!
-
-    ```javascript
-    function Jedi() {
-      console.log('new jedi');
-    }
-
-    // bad
-    Jedi.prototype = {
-      fight: function fight() {
-        console.log('fighting');
-      },
-
-      block: function block() {
-        console.log('blocking');
+    class Jedi {
+      constructor(options = {}) {
+        const lightsaber = options.lightsaber || 'blue';
+        this.set('lightsaber', lightsaber);
```

Comment:
Getters and setters can be used here. However, I personally have a strong opinion against them.

It's possible to write code that does:

```js
class Jedi {
  constructor(options = {}) {
    this.lightsaber = options.lightsaber || 'blue';
  }

  get lightsaber() {
    return (this.team === 'rebel') ? 'blue' : 'red';
  }

  set lightsaber(ls) {
    while (true) { }
  }
}
```

which will then fail when you do:

```js
const jedi = new Jedi();
jedi.lightsaber = 'blue';
```

While the same (writing really stupid code) is true for methods, at least the methods are explicit (when read) in that an action will take place when they're called. 

Having side effects happen when you set a method on a class is a code smell IMO -- just side effects in general are meh.

That said, thanks for pointing out this example. Setting properties on `this` using strings is a little strange. Ideally we'd have JS be strongly typed, not stringly typed o_O.